### PR TITLE
Alerting: Update Email receiver to parse settings using encoding/json

### LINKED ISSUE #55139: Alerting: Notifier settings do not accept "" as equivalent to nil/missing, allowing creation of dead notifiers
**What happened**:

[Prometheus notifier settings](https://github.com/prometheus/alertmanager/blob/3d624c0552e173aa57bbdad52a55788e38744252/config/notifiers.go#L348) all behave with `"" = nil = <missing>`. There is no semantic difference between empty/nil/missing values.

Almost all of the Grafana APIs work this way as well. However, the alerting notifier settings blob breaks this convention. Almost none of our notifiers handle `""` in an inconsistent way and will fail to use the correct default values.

Grafana notifiers will allow empty strings to bypass validation. This means you can make normally required fields empty, and the notifier will not be rejected. For non-required fields, Grafana will also fail to substitute the correct default value. This allows users to create invalid "dead" notifiers via the Alerting API.

For example, one can create a dead Slack notifier by sending `endpointUrl=""`. The resulting notifier will fail to fall back to the default of `https://slack.com/api/chat.postMessage` even though the URL cannot be empty.

Shout out to @yuri-tceretian for bringing this to my attention!

**What you expected to happen**:

I expected the `settings` block to work under `"" = nil = <missing>` and correctly validate fields. It should return a 400 if required fields are `""` or otherwise substitute correct defaults. This applies to all notifiers.

**How to reproduce it (as minimally and precisely as possible)**:

1. Create a Slack notifier with `endpointUrl = ""`.
2. It's accepted by the API
3. Route notifications through it
4. Nothing delivered

**Anything else we need to know?**:

**Environment**:
- Grafana version: main
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What this PR does / why we need it**:

Instead of using `simplejson` to loosely parse notifier settings which is [bug-prone](https://github.com/grafana/grafana/issues/55139), this PR instead makes notifiers declare a well-defined struct for their settings.

This has some benefits:
- Improves validation for all settings fields and eliminates bugs (see attached issue)
- The total set of fields for each notifier is now clear and obvious. Settings are easier to reason about.
- Simplifies notifier construction logic by removing a layer of indirection
- Eliminates repetition/mapping of settings fields, which was a vector for bugs
- Makes our notifiers more in-line with [Prometheus's notifiers](https://github.com/prometheus/alertmanager/blob/3d624c0552e173aa57bbdad52a55788e38744252/notify/slack/slack.go). They also use settings structs.

**Which issue(s) this PR fixes**:

Fixes: https://github.com/grafana/grafana/issues/55139

### CODE REVIEW

File: pkg/services/ngalert/notifier/channels/email.go [hunk 1]
```diff
@@ -157,3 +165,13 @@ func (en *EmailNotifier) Notify(ctx context.Context, alerts ...*types.Alert) (bo
...
+
+func splitEmails(emails string) []string {
+	return strings.FieldsFunc(emails, func(r rune) bool {
+		switch r {
+		case ',', ';', '\n':
+			return true
+		}
+		return false
+	})
+}
```
Comment (yuri-tceretian) [inline pkg/services/ngalert/notifier/channels/email.go:177]: this is copied from "github.com/grafana/grafana/pkg/util"

**Review by alexmobo — APPROVED** [review]
LGTM

File: pkg/services/ngalert/notifier/channels/email.go [hunk 2]
```diff
@@ -2,96 +2,104 @@ package channels
...
-		Base:        channels.NewBase(config.NotificationChannelConfig),
-		Addresses:   config.Addresses,
-		SingleEmail: config.SingleEmail,
-		Message:     config.Message,
-		Subject:     config.Subject,
-		log:         l,
-		ns:          ns,
-		images:      images,
-		tmpl:        t,
+func buildEmailNotifier(fc channels.FactoryConfig) (*EmailNotifier, error) {
```
Comment (santihernandezc) [inline pkg/services/ngalert/notifier/channels/email.go:82]: Either we change the name of this function to `newEmailNotifier` and un-capitalize the function name in the comment, or we update the comment above to mention the correct function name. I'm leaning towards the first option, but that's up to you.

File: pkg/services/ngalert/notifier/channels/testing.go [hunk 3]
```diff
@@ -86,6 +86,10 @@ type emailSender struct {
 	ns *notifications.NotificationService
 }
 
+func (e emailSender) SendWebhook(ctx context.Context, cmd *channels.SendWebhookSettings) error {
+	panic("not implemented")
```
Comment (santihernandezc) [inline pkg/services/ngalert/notifier/channels/testing.go:90]: Why panicking here? Wouldn't it be better to just return an error?

**Comment by yuri-tceretian** [comment]
Closing because another PR that includes chages in this one has been merged

File: pkg/services/ngalert/notifier/channels/email.go (hunk 2 above)
Comment (yuri-tceretian) [reply pkg/services/ngalert/notifier/channels/email.go:82]: this follows the common pattern we chose when we started refactoring notifiers.

File: pkg/services/ngalert/notifier/channels/testing.go (hunk 3 above)
Comment (yuri-tceretian) [reply pkg/services/ngalert/notifier/channels/testing.go:90]: Because this is the email stub for email sender.

**Pull request closed without merging**